smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json

```diff
@@ -0,0 +1,76 @@
+{
+  "modulesContent": {
+    "$edgeAgent": {
+      "properties.desired": {
+        "schemaVersion": "1.0",
+        "runtime": {
+          "type": "docker",
+          "settings": {
+            "minDockerVersion": "v1.25",
+            "loggingOptions": "",
+            "registryCredentials": {
+              "buildcr": {
+                "username": "<CR.Username>",
+                "password": "<CR.Password>",
+                "address": "edgebuilds.azurecr.io"
+              }
+            }
+          }
+        },
+        "systemModules": {
+          "edgeAgent": {
+            "type": "docker",
+            "settings": {
+              "image": "edgebuilds.azurecr.io/microsoft/azureiotedge-agent:<Build.BuildNumber>-linux-amd64",
+              "createOptions": ""
+            }
+          },
+          "edgeHub": {
+            "type": "docker",
+            "settings": {
+              "image": "edgebuilds.azurecr.io/microsoft/azureiotedge-hub:<Build.BuildNumber>-linux-amd64",
+              "createOptions": "{\"HostConfig\": {\"PortBindings\": {\"8883/tcp\": [{\"HostPort\": \"8883\"}],\"443/tcp\": [{\"HostPort\": \"443\"}],\"5671/tcp\": [{\"HostPort\": \"5671\"}]}}}"
+            },
+            "status": "running",
+            "restartPolicy": "always"
+          }
+        },
+        "modules": {
+          "tempSensor": {
+            "version": "1.0",
+            "type": "docker",
+            "status": "running",
+            "restartPolicy": "always",
+            "settings": {
+              "image": "edgebuilds.azurecr.io/microsoft/azureiotedge-simulated-temperature-sensor:<Build.BuildNumber>-linux-amd64",
+              "createOptions": ""
+            }
+          },
+          "tempFilter": {
+            "version": "1.0",
+            "type": "docker",
+            "status": "running",
+            "restartPolicy": "always",
+            "settings": {
+              "image": "edgebuilds.azurecr.io/microsoft/azureiotedge-temperature-filter:<Build.BuildNumber>-linux-amd64",
+              "createOptions": ""
+            }
+          },
+          
+        }
+      }
+    },
+    "$edgeHub": {
+      "properties.desired": {
+        "schemaVersion": "1.0",
+        "routes": {
+          "TempFilterToCloud": "FROM /messages/modules/tempFilter/outputs/* INTO $upstream",
```

Comment:
Should the routes be something that can be configured? What if different tests need different routes?

Reply:
The idea is to have 1 deployment per test. (we have the default one for TempSensor test).

Reply:
Right, but the template should be across tests, right?

Reply:
I though about that, but that would require our test infraestructure (aka. Quick Start) to be a lot different than today. I agree we can improve a lot in our e2e test. But as of now each E2E Test case will be 1 VSTS task, which will require a full deployment. So, having a single deployment template and changing dynamically would be a lot more complex. So, the name (template) on the file just means that it has parameters (build number and credentials) that needs to be filled from Script. This is very similar to what we are doing today on Stress/Long Haul test. 

Ideally, we should be using a test framework, where we can have 1 single Setup, 1 single tead down and each test case would be just a set of TestSetup (changing deployment)  and a Test Verify... 

---
In reply to: [209095732](https://github.com/Azure/iotedge/pull/140#discussion_r209095732) [](ancestors = 209095732)